Don't resize on opening media and switching modes

uiryuu opened the pull request:
- [x] I have read [CONTRIBUTING.md](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md)
---
**Description:**
Thanks to https://github.com/iina/iina/pull/5029, now initiating rendering doesn't require a forced draw to the main window. Previously, IINA first draw the video in a small window, then resize to the desired window size. This PR change the behavior to show the main window after the geometry is calculated, therefore no resizing animation is required, the window will show directly to the final size.

Specifically, this PR:
- Removed `windowWillOpen()` and moved contents to newly added `override showWindow(_:)`
- Removed `windowDidOpen()` and moved contents to the end of `windowDidLoad()`
  - `PlayerWindowController.windowDidOpen()` also got removed.
- Removed code which explicitly shows the main window when opening media and creating the main window
- Calls `showWindow(_:)` after the final window geometry is calculated and set as the window frame
- Show mini player after videoreconfig

`windowWillOpen()` and `windowDidOpen()` are not standard NSWindow functions, and they are easy to be confused with other NSWindow lifecycle functions, so I moved them and move the code into override functions.

The current implementation is far from clean, but in order to not refactor too much code, I only made changes to necessary parts. Also I feel the animation for changing modes (music -> normal and reverse) bit clunky and not consistent, but that's not in the scope of this PR.

**Demo:**
Before:
https://github.com/iina/iina/assets/20237141/241eca90-66b5-47e9-92c1-551316fc1e36

After:
https://github.com/iina/iina/assets/20237141/3ea7629e-38b2-47ba-ba5c-9c36c9707990

uiryuu committed:
Don't resize when opening a media

This commit:
- Removed `windowWillOpen()` and moved contents to newly added `override
  showWindow(_:)`
- Removed `windowDidOpen()` and moved contents to the end of
  `windowDidLoad()`
  - `PlayerWindowController.windowDidOpen()` also got removed.
- Removed code which explicitly shows the main window when opening media
  and creating the main window
- Calls `showWindow(_:)` after the final window geometry is calcuated and
  set as the window frame

Handle entering music mode correctly

+ cleared up the code

uiryuu committed:
Show mini player after video reconfig

low-batt reviewed (approved):
Pulled PR, built IINA, tested. Looks good to me.

svobs commented:
Looks like some much needed refactoring. Although I for one began to enjoy the zoom animation 😂

uiryuu commented:
Known issues: when opening online resources, the window will still resize after the buffering is done. We need a UI to reflect the buffering process.

lhc70000 commented:
Try this:

- Enable auto switching to music mode (enabled by default)
- Open a music file, IINA opens in music mode
- Click the back button to return to normal mode

Expected: The main window appears with the album art.
Actual: No video window shown, the welcome window is shown instead. The player hangs when trying to quit IINA.

lhc70000 commented:
@svobs I actually feel nostalgic about the animation. Maybe I'm going to implement a more sleek opening animation and make it an option.

uiryuu committed:
Fixed an issue where window doesn't show up while playing audio files

lhc70000 commented on iina/MainWindowController.swift:
```diff
@@ -2584,7 +2570,10 @@ class MainWindowController: PlayerWindowController {
-      if player.disableWindowAnimation || Preference.bool(for: .disableAnimations) {
+      if !player.mainWindow.window!.isVisible && !player.isInMiniPlayer {
+        window.setFrame(rect, display: false, animate: false)
+        showWindow(nil)
```
Why put it here in `adjustFrameByVideoSize()`? This can be called whenever the video's size is changed.

Consider the following case: I open an IINA window and hide it using Cmd+H. Then it automatically continues to the next video (or some plugin/mpv script changed the video/window size), and this line will show the window unexpectedly.

uiryuu committed:
Add a flag to indicate the window is waiting for size

- Changed the method for detecting no album art in mini player
- Renamed `notifyMainWindowVideoSizeChanged()` to
  `notifyWindowVideoSizeChanged()`
- Don't try to update video size on tracklistchange, instead, wait for videoreconfig

uiryuu committed:
Set pendingShow flag even if `showMainWindow = false` for a future video
reconfig event

uiryuu committed:
Fix sometimes mini player is not key

uiryuu committed:
Added comments, fix a bug where switching back from mini player
resulting black screen

uiryuu committed:
Make mini player and main window symmetric

- Merge `adjustFrameByVideoSize` and `updateVideoSize` into
  `handleVideoSizeChange`
- Move `pendingShow` into `PlayerWindowController`
- Set `pendingShow` for both player to true when loading files

low-batt commented:
This was included in IINA 1.4.0.
